mmseg/models/backbones/hrformer.py

```diff
@@ -0,0 +1,740 @@
+            head_dim ** -0.5 if set. Default: None.
+        attn_drop_rate (float, optional): Dropout ratio of attention weight.
+            Default: 0.0
+        proj_drop_rate (float, optional): Dropout ratio of output. Default: 0.
+        with_rpe (bool, optional): If True, use relative position bias.
+            Default: True.
+        init_cfg (dict | None, optional): The Config for initialization.
+            Default: None.
+    """
+
+    def __init__(self,
+                 embed_dims,
+                 num_heads,
+                 window_size,
+                 qkv_bias=True,
+                 qk_scale=None,
+                 attn_drop_rate=0.,
+                 proj_drop_rate=0.,
+                 with_rpe=True,
+                 init_cfg=None):
+
+        super().__init__(init_cfg=init_cfg)
+        self.embed_dims = embed_dims
+        self.window_size = window_size  # Wh, Ww
+        self.num_heads = num_heads
+        head_embed_dims = embed_dims // num_heads
+        self.scale = qk_scale or head_embed_dims**-0.5
+
+        self.with_rpe = with_rpe
+        if self.with_rpe:
+            # define a parameter table of relative position bias
+            self.relative_position_bias_table = nn.Parameter(
+                torch.zeros(
+                    (2 * window_size[0] - 1) * (2 * window_size[1] - 1),
+                    num_heads))  # 2*Wh-1 * 2*Ww-1, nH
+
+            # pairwise relative position for each token inside the window
+            coords_h = torch.arange(self.window_size[0])
+            coords_w = torch.arange(self.window_size[1])
+            # [2, Wh, Ww]
+            coords = torch.stack(torch.meshgrid([coords_h, coords_w]))
+            # [2, Wh*Ww]
+            coords_flatten = torch.flatten(coords, 1)
+            # [2, Wh*Ww, Wh*Ww]
+            relative_coords = coords_flatten[:, :, None] - \
+                coords_flatten[:, None, :]
+            # [Wh*Ww, Wh*Ww, 2]
+            relative_coords = relative_coords.permute(1, 2, 0).contiguous()
+            # shift the values to make them start from 0
+            relative_coords[:, :, 0] += self.window_size[0] - 1
+            relative_coords[:, :, 1] += self.window_size[1] - 1
+            relative_coords[:, :, 0] *= 2 * self.window_size[1] - 1
+            # [Wh*Ww, Wh*Ww]
+            relative_position_index = relative_coords.sum(-1)
+            self.register_buffer('relative_position_index',
+                                 relative_position_index)
+
+        self.qkv = nn.Linear(embed_dims, embed_dims * 3, bias=qkv_bias)
+        self.attn_drop = nn.Dropout(attn_drop_rate)
+        self.out_proj = nn.Linear(embed_dims, embed_dims)
+        self.proj_drop = nn.Dropout(proj_drop_rate)
+
+        self.softmax = nn.Softmax(dim=-1)
+
+    def init_weights(self):
+        super(WindowMSA, self).init_weights()
+        trunc_normal_init(self.relative_position_bias_table, std=0.02)
```

Comment:
might add if condition
```suggestion
        if self.with_rpe:
            trunc_normal_init(self.relative_position_bias_table, std=0.02)
```